ext/semian/simple_integer.c

```diff
@@ -0,0 +1,90 @@
+#include "simple_integer.h"
+
+#include <errno.h>
+#include <sys/ipc.h>
+#include <sys/shm.h>
+#include "types.h"
+#include "util.h"
+
+static const rb_data_type_t semian_simple_integer_type;
+
+static int* get_value(VALUE self) {
+  const char* name = to_s(rb_iv_get(self, "@name"));
+  if (name == NULL) {
+    rb_raise(rb_eArgError, "could not get object name");
+  }
+
+  key_t key = generate_key(name);
+
+  const int permissions = 0664;
+  int shmid = shmget(key, sizeof(int), IPC_CREAT | permissions);
+  if (shmid == -1) {
+    rb_raise(rb_eArgError, "could not create shared memory (%s)", strerror(errno));
+  }
+
+  void *val = shmat(shmid, NULL, 0);
+  if (val == (void*)-1) {
+    rb_raise(rb_eArgError, "could not get shared memory (%s)", strerror(errno));
+  }
+
+  return (int*)val;
+}
```

Comment:
`@name` doesn't look to be changing after initialization. Could we cache the `shmid` in `semian_simple_integer_t`? How about the `void *val`?

Similarly for `circuit_breaker.c` and `sliding_window.c`.

Reply:
Yes, I actually addressed this in a later commit.
Basically, we want to store the `key` generated from the name on the object itself.
I'll double check I've done this everywhere.